src/IO/BrotliWriteBuffer.h

```diff
@@ -22,6 +22,7 @@ class BrotliWriteBuffer : public BufferWithOwnMemory<WriteBuffer>
 
 private:
     void nextImpl() override;
+    void finalize() override { out->finalize(); }
```

Comment:
Fixed in #21305 (and in a little bit better way), same for all other writers

Reply:
And BTW finalize() is a public part of interface (and this is a why it does not shows the conflicts partially)

Reply:
This is a good fix as well, but as I see write buffers shall pass `finalize()` to underlying write buffer anyway.

Reply:
Indeed, there should be finalize(), not next() (here https://github.com/ClickHouse/ClickHouse/blob/15b3f379a56c8f648cff38695ec2aa617c13fe58/src/IO/BrotliWriteBuffer.cpp#L109 and in all similar places).

Can you send a separate patch for this?

Reply:
I do not agree, and `finish()` there shall be called from `finalize()`, not a destructor.

Reply:
I see, we shall call `finalize()` of underlying buffer from `finish()`, but `finish()` shall be the same thing as finalize actually.

Reply:
#22147